main/.vitepress/enhanceApp.js

```diff
@@ -1,3 +1,7 @@
+// XXX
+// vitepress does not seem to expose the vue router directly, so we'll
+// have to refactor this. some redirects are working in themeConfig/rewrites.js
+
```

Comment:
@gibson042 when you have some time, could you take a look at this PR and help me understand #882 and #888 better? 

For #888, external redirects, is it sufficient to use `_redirects` and handle this outside the application? [Cloudflare Pages Redirects](https://developers.cloudflare.com/pages/configuration/redirects/)

For #882, I'm not sure I understand the motivation enough to tell if the issue is still persistent with vitepress.

_If you are looking for hosted preview URL, try this: https://0ebcfb0d.documentation-7tp.pages.dev/_

Reply:
Let's check #888 first:
* https://0ebcfb0d.documentation-7tp.pages.dev/guides/js-programming/ses/lockdown.html is good ✔
* https://0ebcfb0d.documentation-7tp.pages.dev/zoe/guide/contracts/covered-call.html is not ✘

I think the latter can be fixed with [greedy matching](https://developers.cloudflare.com/pages/configuration/redirects/#splats), e.g.
```
/zoe/guide /guides/zoe/
/zoe/guide/* /guides/zoe/:splat
```

So probably we just want to map `redirects` from **main/.vuepress/enhanceApp.js** into **_redirects**:
* Each prefix-style entry in the former should become two entries in the latter as above.
* Each leaf-page entry in the former should become two entries in the latter to the same destination, one with a source ending in ".html" and the other not (e.g. https://0ebcfb0d.documentation-7tp.pages.dev/wallet-api.html and https://0ebcfb0d.documentation-7tp.pages.dev/wallet-api should both work).
* For proper behavior, at least all non-external redirects should use status code [301 (Moved Permanently)](https://www.rfc-editor.org/rfc/rfc9110.html#name-301-moved-permanently).

----

As for #882, it looks like we do need to carry the kludge forward (or find a way to fix things upstream). Observe how clicking either of the following links in the foreground (e.g., **not** a background tab) scrolls away from the in-page anchor to the top of page:
* https://0ebcfb0d.documentation-7tp.pages.dev/reference/zoe-api/zoe-data-types.html#keywordrecord
* https://0ebcfb0d.documentation-7tp.pages.dev/reference/zoe-api/zoe-data-types.html#amountkeywordrecord

But something is different, because even manual application doesn't appear to be correcting the issue.

Reply:
Thanks @gibson042 ! 

For #888, it turns out the build you were looking at already had the custom scroll logic in place. In [5983738](https://github.com/Agoric/documentation/pull/971/commits/5983738feb8c417b76a23d10b1cc1defdaebcb55) I removed the logic, and scroll to anchor on initial page load seems to work fine (as a nit, it could scroll up a bit more, but it seems to be fine from my POV)


For #882 (redirects), I followed your suggestion in [c69ed8a](https://github.com/Agoric/documentation/pull/971/commits/c69ed8a7c4cb864ca40428bcf489bd4cb3478532):


I am not sure if we need something like: `/ertp/guide/ /guides/ertp/ ` in addition to `/ertp/guide /guides/ertp/`, but it doesn't seem to cause any harm. `/* /:splat` seems to cover the covered-call example you shared. _Verified on this [deploy url](https://65f33b9537ffb20c818e3c7a--visionary-marzipan-3cd6de.netlify.app/)._